### PR TITLE
[NTOS:MM] MiDbgDumpAddressSpace(): Fix end values, Add more entries

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Be consistent and more explicit.

## Proposed changes

Fix end values, Add more entries.

### CODE REVIEW

File: ntoskrnl/mm/mminit.c [hunk 1]
```diff
@@ -127,46 +127,83 @@ MiDbgDumpAddressSpace(VOID)
     //
     DPRINT1("          0x%p - 0x%p\t%s\n",
             KSEG0_BASE,
-            (ULONG_PTR)KSEG0_BASE + MmBootImageSize,
+            (ULONG_PTR)KSEG0_BASE + MmBootImageSize - 1,
```
Comment (HBelusca) [inline ntoskrnl/mm/mminit.c:130]: What if the convention of the displayed ranges were `[start, end)` (or as we write in french maths, `[start, end[`) ?

File: ntoskrnl/mm/mminit.c [hunk 2]
```diff
@@ -127,46 +127,83 @@ MiDbgDumpAddressSpace(VOID)
...
-            PTE_BASE, PTE_TOP,
+            PTE_BASE,
+            PTE_TOP,
             "Page Tables");
     DPRINT1("          0x%p - 0x%p\t%s\n",
-            PDE_BASE, PDE_TOP,
-            "Page Directories");
+            PDE_BASE,
+            PDE_TOP,
+            "Page Directories (inside Page Tables)");
```
Comment (HBelusca) [inline ntoskrnl/mm/mminit.c:155]: Actual kernel developers know that fact already.

File: ntoskrnl/mm/mminit.c [hunk 3]
```diff
@@ -127,46 +127,83 @@ MiDbgDumpAddressSpace(VOID)
...
+#ifdef _M_AMD64
     DPRINT1("          0x%p - 0x%p\t%s\n",
-            HYPER_SPACE, HYPER_SPACE_END,
+            PPE_BASE,
+            PPE_TOP,
+            "PPE (inside Page Directories)");
```
Comment (HBelusca) [inline ntoskrnl/mm/mminit.c:160]: Ditto

File: ntoskrnl/mm/mminit.c [hunk 4]
```diff
@@ -127,46 +127,83 @@ MiDbgDumpAddressSpace(VOID)
...
+    DPRINT1("          0x%p - 0x%p\t%s\n",
+            PXE_BASE,
+            PXE_TOP,
+            "PXE (inside PPE)");
```
Comment (HBelusca) [inline ntoskrnl/mm/mminit.c:164]: Ditto

File: ntoskrnl/mm/mminit.c [hunk 5]
```diff
@@ -127,46 +127,83 @@ MiDbgDumpAddressSpace(VOID)
...
             "Non Paged Pool Expansion PTE Space");
+    DPRINT1("          0x%p - 0x%p\t%s\n",
+            MI_DEBUG_MAPPING,
+            (ULONG_PTR)MI_DEBUG_MAPPING + PAGE_SIZE - 1,
+            "Debugger mapping");
+#ifdef _X86_
+    DPRINT1("          0x%p - 0x%p\t%s\n",
+            MM_HAL_VA_START,
+            MM_HAL_VA_END,
+            "Reserved HAL area (includes KUSER_SHARED_DATA and KPCR)");
```
Comment (HBelusca) [inline ntoskrnl/mm/mminit.c:194]: Ditto

File: ntoskrnl/mm/mminit.c (hunk 1 above)
Comment (SergeGautherie) [reply ntoskrnl/mm/mminit.c:130]: For example, 'end included' avoids the `0xFFFFFFFF + 1` overflow.
Nonetheless, I agree 'end excluded' could be good enough.
The main point is not to mix the two as current code does.

**Comment by tkreuzer** [comment]
This doesn't fix anything, it just changes stuff for the sake of changing it and adds more useless debug prints. Waste of time.

**Pull request closed without merging**